File: neetcode/11. Container With Most Water/11.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea = 0;

        while (left < right) {
            int minHeight = Math.min(height[left], height[right]);
            maxArea = Math.max((right - left) * minHeight, maxArea);

            if (height[left] < height[right]) {
                left++;
            } else if (height[left] > height[right]) {
                right--;
            } else {
                left++;
                right--;
            }
        }

        return maxArea;
    }
}

File: neetcode/11. Container With Most Water/11.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var left = 0
        var right = height.size - 1
        var maxArea = 0

        while (left < right) {
            val minHeight = min(height[left], height[right])

            maxArea = max(maxArea, minHeight * (right - left))

            while (left < right && height[left] <= minHeight) {
                left++
            }

            while (left < right && height[right] <= minHeight) {
                right--
            }
        }

        return maxArea
    }
}
